crates/cxx-qt-gen/include/cxxqt_signalhandler.h

```diff
@@ -0,0 +1,52 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#pragma once
+
+#include <type_traits>
+
+#include "rust/cxx.h"
+
+namespace rust::cxxqtlib1 {
+
+template<typename CXXArguments>
+class SignalHandler
```

Comment:
Maybe add a comment here that this represents a Rust `Box<dyn FnMut>` by deferring all actual operations to Rust, apart from moving it, which is done in terms of raw memory copy, which is equivalent to what Rust does.